Type errors preventing compilation

There are some type errors that show up in my IDE (VS Code) and prevent compiling the library using `tsc`:

<details><summary>Text version</summary>
<pre><code>
> npx tsc
Reader.tsx:254:27 - error TS2532: Object is possibly 'undefined'.

254       const startOffset = virtualizer.getOffsetForIndex(
                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
255         area.pageIndex,
    ~~~~~~~~~~~~~~~~~~~~~~~
256         "start",
    ~~~~~~~~~~~~~~~~
257       )[0];
    ~~~~~~~

useVirtualizerVelocity.ts:22:27 - error TS18047: 'currentScrollOffset' is possibly 'null'.

22       const newVelocity = currentScrollOffset - lastScrollOffset;
                             ~~~~~~~~~~~~~~~~~~~

useVirtualizerVelocity.ts:30:27 - error TS2345: Argument of type 'number | null' is not assignable to parameter of type 'SetStateAction<number>'.
  Type 'null' is not assignable to type 'SetStateAction<number>'.

30       setLastScrollOffset(currentScrollOffset);
                             ~~~~~~~~~~~~~~~~~~~

useZoom.ts:19:5 - error TS2345: Argument of type 'number | null' is not assignable to parameter of type 'number'.
  Type 'null' is not assignable to type 'number'.

19     virtualizer.scrollOffset,
       ~~~~~~~~~~~~~~~~~~~~~~~~

useZoom.ts:24:5 - error TS18047: 'virtualizer.scrollOffset' is possibly 'null'.

24     virtualizer.scrollOffset - currentVirtualItem.index * VIRTUAL_ITEM_GAP;
       ~~~~~~~~~~~~~~~~~~~~~~~~

useZoom.ts:24:32 - error TS18048: 'currentVirtualItem' is possibly 'undefined'.

24     virtualizer.scrollOffset - currentVirtualItem.index * VIRTUAL_ITEM_GAP;
                                  ~~~~~~~~~~~~~~~~~~

useZoom.ts:31:33 - error TS18048: 'currentVirtualItem' is possibly 'undefined'.

31     newScaledOffsetWithoutGap + currentVirtualItem.index * VIRTUAL_ITEM_GAP;
                                   ~~~~~~~~~~~~~~~~~~


Found 7 errors in 3 files.

Errors  Files
     1  Reader.tsx:254
     2  useVirtualizerVelocity.ts:22
     4  useZoom.ts:19
</code></pre>
</p>
</details> 




![Screenshot 2024-11-25 at 15 35 44](https://github.com/user-attachments/assets/ea066c44-ec77-4429-a30d-f0d9ade3574a)

--- a/src/Reader.tsx
+++ b/src/Reader.tsx
@@ -251,10 +251,13 @@ const Reader = ({
     };
 
     const jumpToHighlightArea = (area: HighlightArea) => {
-      const startOffset = virtualizer.getOffsetForIndex(
+      const startOffset = virtualizer.getOffsetForIndex?.(
         area.pageIndex,
         "start",
-      )[0];
+      )?.[0];
+
+      if (startOffset == null) return;
+
       const itemHeight = estimateSize(area.pageIndex);
       const offset = getOffsetForHighlight({
         ...area,

--- a/src/useVirtualizerVelocity.ts
+++ b/src/useVirtualizerVelocity.ts
@@ -18,6 +18,10 @@ const useVirtualizerVelocity = ({
       // Get the current scroll offset
       const currentScrollOffset = virtualizer.scrollOffset;
 
+      if (currentScrollOffset == null) {
+        return;
+      }
+
       // Calculate the difference (velocity)
       const newVelocity = currentScrollOffset - lastScrollOffset;
 

--- a/src/useZoom.ts
+++ b/src/useZoom.ts
@@ -10,7 +10,9 @@ export const calculateAfterZoomOffset = (
   currentScale: number,
   nextScale: number,
 ): number | null => {
-  if (!currentScale || !virtualizer) return null;
+  if (!currentScale || !virtualizer || virtualizer.scrollOffset == null) {
+    return null;
+  }
 
   const offsetScaleFactor = nextScale / currentScale;
 
@@ -18,6 +20,8 @@ export const calculateAfterZoomOffset = (
   const currentVirtualItem = virtualizer.getVirtualItemForOffset(
     virtualizer.scrollOffset,
   );
+
+  if (!currentVirtualItem) return null;
 
   // Subtract off gaps from the current total offset
   const currentOffsetWithoutGap =

